### ISSUE TITLE
Performance/RedundantMerge raises error on some double-splat cases

### ISSUE THREAD

**Issue by jlfaber** [bbatsov/rubocop#4366]
When running `merge!` on a catch-all function argument, passing a double-splatted hash causes rubocop to raise `undefined method 'source' for nil:NilClass`

--------

## Expected behavior

Rubocop should not raise an error.

## Actual behavior

Error in `RuboCop::Cop::Performance::RedundantMerge#to_assignments`.

Stacktrace:
```
An error occurred while Performance/RedundantMerge cop was inspecting test2.rb:4:2.
undefined method `source' for nil:NilClass
rubocop-0.48.1/lib/rubocop/cop/performance/redundant_merge.rb:79:in `block in to_assignments'
rubocop-0.48.1/lib/rubocop/cop/performance/redundant_merge.rb:74:in `map'
rubocop-0.48.1/lib/rubocop/cop/performance/redundant_merge.rb:74:in `to_assignments'
rubocop-0.48.1/lib/rubocop/cop/performance/redundant_merge.rb:24:in `block in on_send'
rubocop-0.48.1/lib/rubocop/cop/performance/redundant_merge.rb:53:in `block in each_redundant_merge'
rubocop-0.48.1/lib/rubocop/cop/performance/redundant_merge.rb:18:in `redundant_merge'
rubocop-0.48.1/lib/rubocop/cop/performance/redundant_merge.rb:46:in `each_redundant_merge'
rubocop-0.48.1/lib/rubocop/cop/performance/redundant_merge.rb:23:in `on_send'
```
## Steps to reproduce the problem

```ruby
def bar(**opts)
  opts.merge!(**foo)
  p opts
end
```

## RuboCop version

```
$ rubocop -V
0.48.1 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-darwin16)
```

## Ruby version

```
$ ruby -v
ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-darwin16]
```
